pex/resolver.py

```diff
@@ -737,7 +737,9 @@ def resolve(
       distribution compatibility. Defaults to the current interpreter.
     :type interpreter: :class:`pex.interpreter.PythonInterpreter`
     :keyword str platform: The exact target platform to resolve distributions for. If ``None`` or
-      ``'current'``, resolve for distributions appropriate for `interpreter`.
+      ``'current'``, resolve for distributions appropriate for `interpreter` instead. If any
+      distributions need to be built, the corresponding interpreter must be provided in the
```

Comment:
> ... If any distributions need to be built, the corresponding interpreter must be provided in the

This is not right. The platforms and interpreters argumenst are _completely_ independent. IOW, if breaking the resolve API was deemed OK, we could collapse the two interpreters and platforms arguments to one distribution_targets argument since that's exactly what's done internally:
https://github.com/pantsbuild/pex/blob/e67412c31648f6e0d0f2a68b9d38dffabb258ebc/pex/resolver.py#L958-L975

Reply:
If that all makes sense, perhaps you mean to say:
```
If any distributions need to be built, use the interpreters argument instead, providing the corresponding interpreter
```

Reply:
Ok. I was under the impression that an interpreter of the appropriate version was necessary to build any non-wheel for that interpreter version. However, now, IIUC, the `interpreter` from this section: https://github.com/pantsbuild/pex/blob/e67412c31648f6e0d0f2a68b9d38dffabb258ebc/pex/resolver.py#L515-L523
will just resolve to the current interpreter with `PythonInterpreter.get()` whenever the target was created with `DistributionTarget.from_platform()`.

I see now that we explicitly error in the case that we are forced into building wheels from a foreign platform. I didn't realize how that was handled before: https://github.com/pantsbuild/pex/blob/e67412c31648f6e0d0f2a68b9d38dffabb258ebc/pex/pip.py#L202-L212

So things learned:
1. no special interpreter is needed to build non-wheels!
2. if there are no wheels available for a foreign platform, we will error with `Unsatisfiable`, because our pip invocation will include `--only-binary :all:`.
3. if wheels are turned off and we try to resolve for a foreign platform, we have the very specific error message displayed above.

I will look again and see whether there is any useful change I can make to the docs. Thank you again for your patience.